### PR TITLE
[luci] Add helper methods for PGraphs

### PR DESCRIPTION
**Pull request by seanshpark**
This will add some helper methods for PGraphs.

ONE-DCO-1.0-Signed-off-by: SaeHie Park <saehie.park@gmail.com>

### CODE REVIEW

**Comment by seanshpark** [comment]
For #6230, from draft #6231

File: compiler/luci/partition/src/PartitionIR.cpp [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+{
+  assert(node != nullptr);
+
+  LOGGER(l);
+
+  auto it = node2group.find(node);
+  if (it == node2group.end())
+    INFO(l) << "PGraphs::group_of " << node << "(" << node->name() << ") not found" << std::endl;
+  assert(it != node2group.end());
+  auto group = it->second;
```
Comment (mhs4670go) [inline compiler/luci/partition/src/PartitionIR.cpp:80]: 1. You intended to just be dereferenced even if `it` is `node2group.end()` in release mode?
2. There is no information about `interest` type here. And neither is the logic that returns a default group when `node` is not `interest` type. Does that logic exist at the place where `node2group` is made?

File: compiler/luci/partition/src/PartitionIR.h [hunk 2]
```diff
@@ -65,8 +65,27 @@ struct PGraphs
...
+   * @note  node maybe one of NOT interest type(like CircleInput), there will be
+   *        no pgraph holding it.
+   */
+  const PGraph *pgraph_of(luci::CircleNode *node) const;
 };
 
 } // namespace luci
 
-#endif // __LUCI_PARTITON_IR_H__
+#endif // __LUCI_PARTITION_IR_H__
```
Comment (seanshpark) [inline compiler/luci/partition/src/PartitionIR.h:91]: I'll post another PR or this...

File: compiler/luci/partition/src/PartitionIR.cpp (hunk 1 above)
Comment (seanshpark) [reply compiler/luci/partition/src/PartitionIR.cpp:80]: You are right. There's seems something missing here.
- we should return empty string if not found
- caller will set to default group if empty
- "interest" is ambigious here. I'll revise the comment

**Review by mhs4670go — APPROVED** [review]
LGTM

File: compiler/luci/partition/src/PartitionIR.h (hunk 2 above)
Comment (seanshpark) [reply compiler/luci/partition/src/PartitionIR.h]: --> #6321

**Review by llFreetimell — APPROVED** [review]

**Pull request merged**